File: Chapter.10-AWS-IoT-Core-With-Lambda/Lambda_Publisher.py
import json
import boto3

client = boto3.client('iot-data', region_name='<Your-AWS-Region-Here>')

def lambda_handler(event, context):
    #event['temperature'] = round(((event['temperature'] - 32) / 1.8))
    print(event)
    response = client.publish(
        topic='fromLambda',
        qos=0,
        payload=json.dumps(event) 
    )
    print(response)
    
    return {
        'statusCode': 200,
        'body': json.dumps('Published to topic')
    }

File: Chapter.10-AWS-IoT-Core-With-Lambda/Lambda_Publisher.js
var AWS = require('aws-sdk');
var iotdata = new AWS.IotData({endpoint: '<YOUR-AWS-IOT-ENDPOINT-HERE>' });

exports.handler = (event) => {
    //event['temperature'] = round(((event['temperature'] - 32) / 1.8))
    console.log("The event object is: " + JSON.stringify(event));
    var params = {
        topic: "fromLambda", 
        payload: JSON.stringify(event), //event.Temperature
        qos: 0
    };

    return iotdata.publish(params, function(err, data) {
        if (err) {
            console.log("If error: " + JSON.stringify(err));
        }
        else {
            console.log("Success");
        }
    })
};

